pydoof/helpers.py

```diff
@@ -42,7 +43,9 @@ def _parse_param(param, value):
         query_params[param] = value.value
     elif not isinstance(value, str) and isinstance(value, Iterable):
         query_params.update(
-            _dicts_appends(_parse_param(f'{param}[]', v) for v in value)
+            _dicts_appends(_parse_param(
+                f'{param}[{i}]' if 'facets' in param else f'{param}[]',
+                v, i) for i, v in enumerate(value))
```

Comment:
I don't think you really need this, I guess you are doing something wrong. Also, you should try to avoid this kind of code. You are changing the parsing for a parameter name when this should be more generic.

Reply:
In general this function seems not be necessary to be touch, as it were it seems ok to me.

Reply:
I do this because facets now should be something like `facets[0]...` instead of `facets[]...`. All the endpoints that use facets should work on this new way.

I throught to avoid this type of practice @mdogo (agreed with you) but If not, I need to get a "wrong" response from this function and then parse it again only for facets. I do this because of performance.

Reply:
> I do this because facets now should be something like facets[0]... instead of facets[].... All the endpoints that use facets should work on this new way.

Why? `facets[0]=f0&facets[1]=f1` is equivalent to `facets[]=f0&facets[]=f1`, and I think we opted for the simpler version.

Reply:
The API and what it is required:
https://redocly.github.io/redoc/?url=https://eu1-search.doofinder.com/swagger_v6.yaml#operation/search_query

The explanation:
https://doofinder.atlassian.net/wiki/spaces/DEV/pages/133890288/Request#Search-endpoint

This works: `facets[0][field]=brand&facets[0][size]=10`.

Reply:
It is mandatory the number at `facets[<number>][field]` because doophoenix this way can read this var as a map and work with it as it expects.

But I think this condition you added here is not necessary, better to do a map of parameters before calling this function so this function doens't have a special condition for one particular parameter

Reply:
Ok, now I got it. But I think you take a wrong approach, it is not about having facets parsed in that way. It is about having a list of maps parsed correctly. Now, the thing is that when you pass an array of objects in the query, and remove the index, like:

```url
?facets[][field]=brand&facets[][size]=10&facets[][field]=category
```
We can't know to which facet that size belongs. Phoenix will take that and parse it as:

```elixir
[%{"field" => "brand"}, %{"size" => "10"}, %{"field" => "category"}]
```

So I don't know what is the best approach, setting the index for all arrays o trying to set the index just for objects arrays...

Damn! That would be so easy with pattern matching... The first one is really simple, but I'm not sure if we are breaking Doophoenix v6 specs with that way. But the second one is not so easy:

```python
def _has_dicts(values):
    return any(isinstance(value, dict) for value in values)

# ...
elif not isinstance(value, str) and isinstance(value, Iterable):
    if _has_dicts(value):
        params = _parse_param(f'{param}[{i}]', v) for i, v in enumerate(value)
    else:
        params = _parse_param(f'{param}[]', v) for v in value)
    query_params.update(
        _dicts_appends(params)
    )
```

Something along these lines